pkg/vgmanager/vgmanager_controller.go

```diff
@@ -269,6 +274,45 @@ func (r *VGReconciler) addThinPoolToVG(vgName string, config *lvmv1alpha1.ThinPo
 	return nil
 }
 
+func (r *VGReconciler) extendThinPool(vgName string, lvSize string, config *lvmv1alpha1.ThinPoolConfig) error {
+
+	vg, err := GetVolumeGroup(r.executor, vgName)
+	if err != nil {
+		if err != ErrVolumeGroupNotFound {
+			return fmt.Errorf("failed to get volume group. %q, %v", vgName, err)
+		}
+		return nil
+	}
+
+	thinPoolSize, err := strconv.ParseFloat(lvSize[:len(lvSize)-1], 64)
+	if err != nil {
+		return fmt.Errorf("failed to parse lvSize. %v", err)
+	}
+
+	vgSize, err := strconv.ParseFloat(vg.VgSize[:len(vg.VgSize)-1], 64)
+	if err != nil {
+		return fmt.Errorf("failed to parse vgSize. %v", err)
+	}
+
+	// return if thinPoolSize is as expected
+	if config.SizePercent == int((thinPoolSize/vgSize)*100) {
```

Comment:
this should be <= instead of ==
we don't want to run lvextend if the thinpoolsize is larger than config.SizePercent (I hope we never get there though ;) )

Reply:
done